services/ux-backend/handlers/onboardingtokens/handler.go

```diff
@@ -52,3 +58,40 @@ func handleUnsupportedMethod(w http.ResponseWriter, r *http.Request) {
 		klog.Errorf("failed write data to response writer: %v", err)
 	}
 }
+
+func getAllocatedQuotaInGib(r *http.Request) (*int, error) {
+	var quota = struct {
+		StorageQuota int    `json:"storageQuota"`
+		Unit         string `json:"unit"`
+	}{}
+	err := json.NewDecoder(r.Body).Decode(&quota)
+	if err != nil {
+		// When error is EOF that means request body is empty and quota
+		// is unlimited
+		if err.Error() == "EOF" {
+			return nil, nil
+		}
```

Comment:
But it does not mean has you can get an EOF even if you have a broken JSON, or a partial JSON. 
EOF means you got to the end of the data stream before a complete json have been read.

If you want to check for an empty body, check the length of the bytes slice/array before sending it to the decode function

Reply:
Sure, I can verify that the content length is 0.